packages/marshal/src/marshal.js

```diff
@@ -682,10 +682,22 @@ export { Remotable };
  * @param {T|undefined} [remotable={}] The object used as the remotable
  * @returns {T} remotable, modified for debuggability
  */
-const Far = (farName, remotable = undefined) => {
+export const Far = (farName, remotable = undefined) => {
   const r = remotable === undefined ? {} : remotable;
   return Remotable(`Alleged: ${farName}`, undefined, r);
 };
-
 harden(Far);
-export { Far };
+
+/**
+ * Coerce `func` to a far function that preserves its call behavior.
+ *
+ * @param {string} farName to be used only if `func` is not already a
+ * far function.
+ * @param {(...args: any[]) => any} func
+ */
+export const ToFarFunction = (farName, func) => {
+  if (getInterfaceOf(func) !== undefined) {
+    return func;
+  }
+  return Far(farName, (...args) => func(...args));
```

Comment:
Why not do `return Far(farName, func)`? Is it because you're trying to avoid hardening `func`? If so, you might mention that in the docstring, and any other reason why you might use `ToFarFunction` instead of plain `Far`.

Reply:
I explained in the docstring. 

```js
/**
 * Coerce `func` to a far function that preserves its call behavior.
 * If it is already a far function, return it. Otherwise make and return a
 * new far function that wraps `func` and forwards calls to it. This
 * works even if `func` is already frozen. `ToFarFunction` is to be used
 * when the function comes from elsewhere under less control. For functions
 * you author in place, better to use `Far` on their function literal directly.
```

Resolving.